features/networking/multus.feature

```diff
@@ -170,3 +170,97 @@ Feature: Multus-CNI related scenarios
     #  | curl | --connect-timeout | 5 | <%= cb.pod1_multus_ip %>:8080 |
     #Then the step should succeed
     #And the output should contain "Hello OpenShift"
+
+  # @author bmeng@redhat.com
+  # @case_id OCP-21853
+  @admin
+  Scenario: Create pods with multus-cni - host-device
+    # Make sure that the multus is enabled
+    Given the master version >= "4.0"
+    And the multus is enabled on the cluster
+    And I select a random node
```

Comment:
@bmeng we already have a step `I select a random node's host`, so I don't think we need to define this new step.  We can re-use the exiting step like the follow:
```

    And I select a random node's host
    And evaluation of `host.name` is stored in the :target_node clipboard
```

Reply:
@pruan-rht yeah, as I said in the commend above. The existing step `I select a random node's host` always fail on my 4.0 testing. I think this is caused by the step will try to resolve the hostname and get the IP, but the 4.0 cluster node cannot be resolved. So I copy the existing step and remove the `host` part to make it get node info successfully.

Reply:
@bmeng that's strange, I just tried it and was successful using the original method.  Can you pls paste the error you see here?

Reply:
Here is the error which I met when using the existing step.

```
    And I select a random node's host                                                                                                                                      # features/step_definitions/node.rb:9
      [09:51:15] INFO> Shell Commands: oc get nodes --output=yaml --config=/root/workdir/fedora28-root/ocp4_admin.kubeconfig
      apiVersion: v1
      items:
      - apiVersion: v1
        kind: Node
        metadata:
          annotations:
            machine.openshift.io/machine: openshift-machine-api/qe-bmeng-ca-411-64px5-worker-ca-central-1a-nr4nb
....
....
        resourceVersion: ""
        selfLink: ""

      [09:51:17] INFO> Exit Status: 0
      [09:51:17] INFO> Shell Commands: oc get nodes ip-10-0-141-19.ca-central-1.compute.internal --output=yaml --config=/root/workdir/fedora28-root/ocp4_admin.kubeconfig
      apiVersion: v1
      kind: Node
....
....
          osImage: Red Hat Enterprise Linux CoreOS 410.8.20190329.0 (Ootpa)
          systemUUID: ec2705aa-2366-735d-8f4f-4fc5479b436b

      [09:51:19] INFO> Exit Status: 0
      getaddrinfo: Name or service not known (SocketError)
      /usr/share/ruby/socket.rb:227:in `getaddrinfo'
      /usr/share/ruby/socket.rb:227:in `foreach'
      /usr/share/ruby/socket.rb:631:in `tcp'
      /root/gitrepos/verification-tests-bmeng/lib/ssh.rb:251:in `initialize'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:763:in `new'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:763:in `ssh'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:768:in `exec_raw'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:488:in `mkdir'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:93:in `workdir'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:465:in `exec_as'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:182:in `exec'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:409:in `dns_resolve'
      /root/gitrepos/verification-tests-bmeng/lib/host.rb:424:in `local_ip?'
      /root/gitrepos/verification-tests-bmeng/lib/openshift/node.rb:45:in `block in host'
      /root/gitrepos/verification-tests-bmeng/lib/openshift/node.rb:44:in `each'
      /root/gitrepos/verification-tests-bmeng/lib/openshift/node.rb:44:in `find'
      /root/gitrepos/verification-tests-bmeng/lib/openshift/node.rb:44:in `host'
      /root/gitrepos/verification-tests-bmeng/features/step_definitions/node.rb:13:in `/^I select a random node's host$/'
      features/networking/multus.feature:181:in `And I select a random node's host'
```

Reply:
ok, I think I know what is the problem.
I only added the 3 masters and lb and bastion to the env `OPENSHIFT_ENV_OCP4_HOSTS`, so it will meet problem when it selected a node. After I added all the masters and nodes to the env_var, it works fine now.

Reply:
You don't need to add all nodes. You can copy/paste hosts.spec from flexy job to get it working. Only you need to specify all masters though.

Reply:
so we can use original step now?

Reply:
Hi Alex,
I did not use the flexy job to setup cluster, so I cannot get the `host.spec` directly. I usually combine the nodes' info and export the env by myself...

Reply:
@bmeng, use Flexy...it'll take away your headaches. :)